addons/survey/controllers/user_input_session.py

```diff
@@ -0,0 +1,122 @@
+
+        survey.next_question()
+        template_values = self._prepare_manage_session_values(survey)
+        template_values['is_transitioned'] = True
+        return request.env.ref('survey.user_input_session_manage_content').render(template_values).decode('UTF-8')
+
+    @http.route('/survey/session_results/<string:survey_token>', type='json', auth='user', website=True)
+    def survey_session_results(self, survey_token, **kwargs):
+        """ This route is called when the host shows the current question's results.
+
+        It's not a regular 'request.render' route because we handle the display of results using
+        an AJAX request to be able to include the results in the currently displayed page. """
+
+        survey = self._fetch_from_token(survey_token)
+
+        if not survey.session_state:
+            # no open session
+            return werkzeug.utils.redirect('/')
+
+        user_input_lines = request.env['survey.user_input.line'].search([
+            ('user_input_id', 'in', survey.user_input_ids.ids),
+            ('question_id', '=', survey.session_current_question_id.id)
+        ])
+        question_statistics = survey.session_current_question_id._prepare_statistics(user_input_lines)[0]
+
+        return request.env.ref('survey.survey_page_statistics_question').render({
+            'page_record_limit': 10,
+            'hide_question_title': True,
+            'no_print_url': True,
+            'survey': survey,
+            'question': survey.session_current_question_id,
+            'question_data': question_statistics,
+        }).decode('UTF-8')
+
+    @http.route('/survey/session_ranking/<string:survey_token>', type='json', auth='user', website=True)
+    def survey_session_ranking(self, survey_token, **kwargs):
+        """ This route is called when the host shows the current question's attendees ranking.
+
+        It's not a regular 'request.render' route because we handle the display of the ranking
+        using an AJAX request to be able to include the results in the currently displayed page. """
+
+        survey = self._fetch_from_token(survey_token)
+
+        if not survey.session_state:
+            # no open session
+            return werkzeug.utils.redirect('/')
+
+        return request.env.ref('survey.user_input_session_ranking').render({
+            'ranking': survey._prepare_ranking_values()
+        }).decode('UTF-8')
+
+    def _prepare_manage_session_values(self, survey):
+        questions = list(enumerate(survey.question_ids))
+        current_question_index = questions.index(
+            next(question for question in questions if question[1] == survey.session_current_question_id)
+        )
+
+        base_url = request.env['ir.config_parameter'].sudo().get_param("web.base.url")
+        return {
+            'answer': request.env['survey.user_input'],
+            'survey': survey,
+            'is_last_question': current_question_index == (len(survey.question_ids) - 1),
```

Comment:
is current_question_index really necessary ? Couldn't this be done with : 
`survey.session_current_question_id == survey.question_ids[len(survey.question_ids)-1]` ?

Reply:
Yes, nice catch.
I've changed a little bit to account for the fact that survey.question_ids could be empty (even though it's not supposed to happen).

Also, did you know that in python you can get the last element by just doing `list[-1]` ? Pretty handy!